memq/src/main/java/com/pinterest/memq/core/rpc/MemqNettyServer.java

```diff
@@ -194,7 +204,9 @@ private MetricRegistry initializeMetrics() throws UnknownHostException {
 
     if (client != null) {
       String localHostname = MiscUtils.getHostname();
-      for (String metricName : registry.getNames()) {
+      List<String> metricsNames = new ArrayList<>(registry.getNames());
+      metricsNames.addAll(DEPENDENCY_METRIC_NAMES);
```

Comment:
Why can't this new metric be added to `registry` so we don't have to manually add it?

Reply:
It has been added to the registry. The list is used to create a reporter for this metric.

You can check the code starting at line 185. There are logics for adding metrics from ByteBufAllocator. Following this, there's a for loop that creates the reporter. While I cannot add this metric directly here, I do want to use the same for loop to create the reporter and avoid duplicate code. That's why I added this new hardcoded list.